### DISCUSSION TITLE
How can we getDirty() from AssignRole()?

### DISCUSSION THREAD

**Discussion by GBry** [spatie/laravel-permission#2004] [Q&A] [answered]
As titled, I need it for logging purposes.. but apparently AssignRole directly saves, so I can't getDirty on model..

**Comment by PaolaRuby** [comment, marked as answer]
Override the method with your own implementation

**Reply by GBry** [reply]
Right. But since I'm new to Laravel and general PHP.. is there a simpler, more beginner-friendly way?

**Comment by GBry** [comment]
Right, so @PaolaRuby nudged me in the right direction. The way is to run a logging function (adding another field in my model) to log the changes..

**Reply by PaolaRuby** [reply]
yes, that was what it said, but you take the credit as answered, that is why people stop answering 😄

**Reply by GBry** [reply]
How can I mark yours as answer?

**Reply by PaolaRuby** [reply]
As same like you do the first time with yourself

**Reply by GBry** [reply]
Thanks! Sorry, I'm new to Github's discussions and issues..
